libsolidity/interface/CompilerStack.cpp

```diff
@@ -1248,6 +1248,15 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	if (!compiledContract.yulIR.empty())
 		return;
 
+	solAssert(_contract.sourceUnit().annotation().useABICoderV2.set(), "");
+	if (!*_contract.sourceUnit().annotation().useABICoderV2)
+		m_errorReporter.warning(
```

Comment:
Should this be some kind of error?

Reply:
What exactly do you mean should be an error?

Reply:
I think it's fine to keep this a warning.